### ISSUE TITLE
scipy 0.16.1-np110py27_p0 [mkl] wants to update the wrong direction

### ISSUE THREAD

**Issue by adamjones1** [conda/conda#2137]
Under python 2.7.11 with mkl, I ran `conda search --outdated`, and noticed that many packages were out of date. I manually updated several which at one point caused a downgrade to 2.7.10, and then upgrade back to 2.7.11 (once this happened once, it didn't do it again).

However, after verifying that outdated showed that I was running the latest packages for everything (limited by mkl), I decided for fun to run `conda update --all` and it gave me the following output:
The following packages will be UPDATED:

    scipy:        0.16.1-np110py27_p0 [mkl] --> 0.16.1-np19py27_p0 [mkl]

The following packages will be DOWNGRADED:

    apptools:     4.3.0-py27_0        --> 4.2.1-py27_0      
    astropy:      1.1.1-np110py27_0   --> 1.0.4-np19py27_0  
    matplotlib:   1.5.1-np110py27_0   --> 1.4.3-np19py27_3  
    numpy:        1.10.2-py27_p0      [mkl] --> 1.9.3-py27_p0      [mkl]
    openssl:      1.0.2f-0            --> 1.0.2d-0          
    pandas:       0.17.1-np110py27_0  --> 0.16.2-np19py27_0 
    path.py:      8.1.2-py27_1        --> 7.6-py27_0        
    pyface:       4.5.2-py27_0        --> 4.4.0-py27_0      
    scikit-learn: 0.17-np110py27_p1   [mkl] --> 0.16.1-np19py27_p0 [mkl]
    traits:       4.5.0-py27_0        --> 4.4.0-py27_0      
    traitsui:     4.5.1-py27_0        --> 4.4.0-py27_0      
    vtk:          6.3.0-py27_1        --> 5.10.1-py27_0 

What I don't understand is -why- it calls a scipy downgrade, an upgrade? (a downgrade which obviously would cause the 10 other packages that I manually upgraded to downgrade as well)

**Comment by jakirkham** [comment]
Yeah, I have seen the python downgrade from 2.7.11 to 2.7.10, but haven't seen this behavior with `scipy` yet.

As to your question of why it is called an upgrade, I think this might be the issue.

    >>> "np19py27" > "np110py27"
    True

Maybe that is the whole issue? Not sure.

**Comment by adamjones1** [comment]
@jakirkham I just did a cursory search of the docs on conda, and I can't find that feature. I know that when I ran `search --outdated` that it showed them in the correct sequential order. Where is this package comparison function?

Also, I guess if it comes down to some input for the comparison being incorrect, how do I (or does someone) notify those who can fix this?

**Comment by jakirkham** [comment]
Similar issue with `openssl`.

```
$ conda update --all
Using Anaconda Cloud api site https://api.anaconda.org
Fetching package metadata: ......
Solving package specifications: ...........................................................................................................................
Package plan for installation in environment /home/kirkhamj/miniconda/envs/nanshenv:

The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    openssl-1.0.2d             |                0         3.2 MB  defaults

The following packages will be DOWNGRADED:

    openssl: 1.0.2f-0 defaults --> 1.0.2d-0 defaults
```

**Comment by jakirkham** [comment]
> I just did a cursory search of the docs on conda, and I can't find that feature.

Not sure what you are asking. Was merely mentioning that the build string might play a role in this problem. Could be a bug in dependency resolution. That is all.

> Where is this package comparison function?

Actually there is a class used for these comparisons. It is called [VersionOrder]( https://github.com/conda/conda/blob/master/conda/version.py#L19 ). I hadn't looked at the code though I know there have been some changes in the solver. It is possible that there was a bug as a result.

> Also, I guess if it comes down to some input for the comparison being incorrect, how do I (or does someone) notify those who can fix this?

That's what the issue tracker is for. :)

**Comment by jakirkham** [comment]
Just in case this is a bug in dependency resolution, figured I should make sure you guys looked at it, @kalefranz @mcg1969.

**Comment by adamjones1** [comment]
@jakirkham You answered my first question, even if you didn't understand what I was asking. Since you only showed what looked like a console output, I figured it was a command-line function. And a quick glance at the docstring tells me all I need to know: version numbering is, in a word, inconsistent.

**Comment by adamjones1** [comment]
Conda 4.0.5 now drops ALL [mkl] packages in favor of the non-[mkl] versions.
